src/com/serotonin/mango/rt/dataImage/PointValueCacheCooperateWithPointValueDao.java

```diff
@@ -0,0 +1,68 @@
+package com.serotonin.mango.rt.dataImage;
+
+import com.serotonin.mango.db.dao.PointValueDao;
+import com.serotonin.mango.rt.dataImage.exceptions_level.ExceptionsArea;
+import java.util.Collections;
+import java.util.LinkedList;
+import java.util.List;
+
+class PointValueCacheCooperateWithPointValueDao {
```

Comment:
If this should be a service and not a dao, you are using dao so this is the layer above.

Reply:
Ok, we can call this class by other name, but this class is as a bridge beetwen cache and dao service.

Reply:
I think it can't be dao. :)